scala-mode.el

```diff
@@ -4,8 +4,8 @@
 
 ;; Homepage: https://github.com/hvesalai/emacs-scala-mode
 ;; Keywords: languages
-;; Package-Version:  0.23
-;; Package-Requires: ()
+;; Package-Version:  1.1.0
+;; Package-Requires: ((emacs "24.2"))
```

Comment:
Why `24.2`? Running `package lint` as `emacs -batch -f package-initialize -f package-lint-batch-and-exit *.el 2>&1 | grep "should depend"` gives me:

```haskell
scala-compile.el:1:49: warning: You should depend on (emacs "24.1") if you need lexical-binding.
scala-compile.el:21:10: error: You should depend on (emacs "24.4") if you need `subr-x'.
scala-compile.el:61:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
scala-compile.el:88:19: error: You should depend on (emacs "24.1") if you need `pcase'.
scala-mode-imenu.el:30:3: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-mapcan'.
scala-mode-imenu.el:34:3: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-mapcan'.
scala-mode-imenu.el:39:22: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-mapcar'.
scala-mode-imenu.el:66:3: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-mapcar'.
scala-mode-imenu.el:70:3: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-destructuring-bind'.
scala-mode-imenu.el:78:12: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-mapcar'.
scala-mode-syntax.el:615:18: error: You should depend on (emacs "24.4") if you need `string-suffix-p'.
scala-mode-syntax.el:846:3: error: You should depend on (emacs "24.1") if you need `syntax-propertize'.
scala-mode-syntax.el:868:3: error: You should depend on (emacs "24.1") if you need `syntax-propertize'.
scala-mode-syntax.el:902:3: error: You should depend on (emacs "24.1") if you need `syntax-propertize'.
scala-organise.el:1:53: warning: You should depend on (emacs "24.1") if you need lexical-binding.
scala-organise.el:15:10: error: You should depend on (emacs "24.4") if you need `subr-x'.
scala-organise.el:47:38: error: You should depend on (emacs "24.4") if you need `string-remove-suffix'.
scala-organise.el:73:7: error: You should depend on (emacs "25.1") or the seq package if you need `seq-find'.
scala-organise.el:83:33: error: You should depend on (emacs "25.1") or the seq package if you need `seq-filter'.
scala-organise.el:86:29: error: You should depend on (emacs "28.1") or the compat package if you need `length='.
```

So the lowest supported version is `28.1`; or if you remove `length=`, it will be `25.1`.

Reply:
Sounds good.  Will you make a new PR to update the requirement?

Reply:
I don't intend to *(I'm kind of hoping people are running more or less modern Emacs — the `28.1` in this case is from 2022. But you never know of course…)*, so I'll leave it to whoever is motivated 😊 

But I'd presume in this case `(length= parts_ 1)` as the only feature from 28.1 may be replaced with `(= (length parts_) 1)` giving us the minimal version of `25.1`.

Reply:
I'll make the PR